### PR TITLE
Circuit for opcode `ADDRESS`

### PR DESCRIPTION
**Pull request by silathdiir**
Spec: https://github.com/privacy-scaling-explorations/zkevm-specs/pull/247

### CODE REVIEW

**Review by DreamWuGit — APPROVED** [review]
LGTM！

File: zkevm-circuits/src/evm_circuit/execution/address.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        block: &Block<F>,
+        _: &Transaction,
+        _: &Call,
+        step: &ExecStep,
+    ) -> Result<(), Error> {
+        self.same_context.assign_exec_step(region, offset, step)?;
+
+        let address = block.rws[step.rw_indices[1]].stack_value();
```
Comment (roynalnaruto) [inline zkevm-circuits/src/evm_circuit/execution/address.rs:73]: Might be worth asserting that the `address` is in fact the call's `callee_address`:

```suggestion
        // you will need `use eth_types::ToAddress;` trait
        let address = block.rws[step.rw_indices[1]].stack_value();
        debug_assert_eq!(call.callee_address, address.to_address());
```

**Review by roynalnaruto — APPROVED** [review]
Looks good! Added a minor point, you may or may not use it

File: zkevm-circuits/src/evm_circuit/execution/address.rs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        block: &Block<F>,
+        _: &Transaction,
+        call: &Call,
+        step: &ExecStep,
+    ) -> Result<(), Error> {
+        self.same_context.assign_exec_step(region, offset, step)?;
+
+        // you will need `use eth_types::ToAddress;` trait
```
Comment (CPerezz) [inline zkevm-circuits/src/evm_circuit/execution/address.rs:73]: I think this comment can be removed. As the trait method is used below.

**Review by CPerezz — APPROVED** [review]
LGTM modulo the comment.

Feel free to remove it and merge later.

File: zkevm-circuits/src/evm_circuit/execution/address.rs (hunk 2 above)
Comment (silathdiir) [reply zkevm-circuits/src/evm_circuit/execution/address.rs:73]: Deleted this useless comment. Thanks.

**Pull request merged**